identity-android/src/main/java/com/android/identity/android/direct_access/DirectAccess.kt

```diff
@@ -0,0 +1,493 @@
+     *
+     * @param documentSlot the slot to certify a credential in
+     * @param credentialData the data being used to certify the credential
+     * @param encryptedPresentationData the data representing the provisioned,
+     *                                  uncertified credential
+     * @return updated encryptedPresentationData
+     */
+    fun certifyCredential(
+        documentSlot: Int,
+        credentialData: ByteArray,
+        encryptedPresentationData: ByteArray
+    ): ByteArray {
+        val bao = ByteArrayOutputStream()
+        try {
+            transport.closeConnection()
+            transport.openConnection()
+            // BEGIN
+            val encryptedData = encryptedPresentationData
+            bao.write(sendApdu(
+                cmd = CMD_MDOC_PROVISION_DATA,
+                slot = documentSlot,
+                data = encryptedData,
+                offset = 0,
+                length = encryptedData.size,
+                operation= PROVISION_BEGIN
+            ))
+
+            // UPDATE
+            val encodedCredData = Cbor.encode(Bstr(credentialData))
+            var remaining = encodedCredData.size
+            var start = 0
+            val maxTransmitBufSize = 512
+            while (remaining > maxTransmitBufSize) {
+                bao.write(sendApdu(
+                    cmd = CMD_MDOC_PROVISION_DATA,
+                    slot = documentSlot,
+                    data = encodedCredData,
+                    offset = start,
+                    length = maxTransmitBufSize,
+                    operation = PROVISION_UPDATE
+                ))
+                start += maxTransmitBufSize
+                remaining -= maxTransmitBufSize
+            }
+
+            // Finish
+            bao.write(sendApdu(
+                cmd = CMD_MDOC_PROVISION_DATA,
+                slot = documentSlot,
+                data = encodedCredData,
+                offset = start,
+                length = remaining,
+                operation = PROVISION_FINISH
+            ))
+            transport.closeConnection()
+        } catch (e: IOException) {
+            transport.closeConnection()
+            throw java.lang.IllegalStateException("Failed to provision credential data $e")
+        }
+
+        // Return updated presentation package.
+        return bao.toByteArray()
+    }
+
+    /**
+     * Returns the number of times the credential in the slot has been used in a presentation since
+     * it was set as the active credential or since the usage count was reset with
+     * [clearCredentialUsageCount].
+     *
+     * @param documentSlot the slot
+     * @return the number of times the credential in the slot was used
+     * @throws Throwable if no active credential
+     */
+    fun getCredentialUsageCount(
+        documentSlot: Int,
+    ): Int {
+        TODO("Not yet implemented")
+    }
+
+    /**
+     * Returns the usage count of the credential (ie. the number of times it was used in a
```

Comment:
nit: This description is accurate, but at first a little confusing and unexpected. Maybe flip the order of operations in this sentence, so the reset comes first?

Or just change this to: "Resets the usage count to 0." And then update the @return documentation to explain that it's the usage count before the reset.